### PR TITLE
Synthetics Info Header Support

### PR DESCRIPTION
**Pull request by TimPansino**
# Overview

* Add support for new `X-NewRelic-Synthetics-Info` header.

### CODE REVIEW

File: newrelic/core/transaction_node.py [hunk 1]
```diff
@@ -76,6 +77,10 @@
         "synthetics_job_id",
         "synthetics_monitor_id",
         "synthetics_header",
```
[the remark this answers is not in the document]
Comment (TimPansino) [reply newrelic/core/transaction_node.py:79]: I don't believe so, it'll be unused.

File: newrelic/core/transaction_node.py [hunk 2]
```diff
@@ -586,6 +591,15 @@ def _event_intrinsics(self, stats_table):
             intrinsics["nr.syntheticsJobId"] = self.synthetics_job_id
             intrinsics["nr.syntheticsMonitorId"] = self.synthetics_monitor_id
 
+        if self.synthetics_type:
+            intrinsics["nr.syntheticsType"] = self.synthetics_type
```
[the remark this answers is not in the document]
Comment (TimPansino) [reply newrelic/core/transaction_node.py:595]: That's handled by the parser logic. `synthetics_type` will be None unless it's a supported version that was handled by the parser.

**Comment by TimPansino** [comment]
> It may be worth adding a test case where version is not 1. In that case, the agent should treat the request like a regular request and not synthetics request.  

There is one, `test_synthetics_event_unsupported_info_version` is the name of that test. It sets the value to 0.

File: tests/agent_features/test_synthetics.py [hunk 3]
```diff
@@ -45,15 +48,19 @@
...
 
-def _make_synthetics_header(
+def _make_synthetics_headers(
     version="1",
     account_id=ACCOUNT_ID,
     resource_id=SYNTHETICS_RESOURCE_ID,
     job_id=SYNTHETICS_JOB_ID,
     monitor_id=SYNTHETICS_MONITOR_ID,
     encoding_key=ENCODING_KEY,
+    info_version="1",
```
[the remark this answers is not in the document]
Comment (TimPansino) [reply tests/agent_features/test_synthetics.py:58]: There is one, test_synthetics_event_unsupported_info_version is the name of that test. It sets the value to 0.

File: tests/testing_support/fixtures.py [hunk 4]
```diff
@@ -793,6 +807,14 @@ def _validate_event_attributes(intrinsics, user_attributes, required_intrinsics,
         assert intrinsics["nr.syntheticsResourceId"] == res_id
         assert intrinsics["nr.syntheticsJobId"] == job_id
         assert intrinsics["nr.syntheticsMonitorId"] == monitor_id
+    
+    if "nr.syntheticsType" in required_intrinsics:
```
[the remark this answers is not in the document]
Comment (TimPansino) [reply tests/testing_support/fixtures.py:811]: Fixed

File: newrelic/core/transaction_node.py [hunk 5]
```diff
@@ -586,6 +591,15 @@ def _event_intrinsics(self, stats_table):
...
+            intrinsics["nr.syntheticsInitiator"] = self.synthetics_initiator
+            if self.synthetics_attributes:
+                # Add all synthetics attributes
+                for k, v in self.synthetics_attributes.items():
+                    if k:
+                        intrinsics["nr.synthetics%s" % camel_case(k)] = v
```
[the remark this answers is not in the document]
Comment (TimPansino) [reply newrelic/core/transaction_node.py:601]: This is actually fine, it's converting to upper camel case. Should read like `nr.syntheticsExampleAttribute`

**Review by umaannamalai — APPROVED** [review]
🚀

**Pull request merged**